crates/test-utils/src/authority.rs

```diff
@@ -106,6 +105,45 @@ where
     handles
 }
 
+/// Spawn all authorities in the test committee into a separate tokio task.
+pub async fn spawn_test_authorities_with_fullnodes<I>(
+    objects: I,
+    config: &NetworkConfig,
+    fullnode_num: u8,
+) -> (Vec<SuiNodeHandle>, Vec<SuiNodeHandle>)
+where
+    I: IntoIterator<Item = Object> + Clone,
+{
+    let mut handles = Vec::new();
+    for validator in config.validator_configs() {
+        let registry_service = RegistryService::new(Registry::new());
+        let node = start_node(validator, registry_service).await;
+        let objects = objects.clone();
+
+        node.with_async(|node| async move {
+            let state = node.state();
+            for o in objects {
+                state.insert_genesis_object(o).await
+            }
+        })
+        .await;
+
+        handles.push(node);
+    }
```

Comment:
this code appears to be copy/pasted from `spawn_test_authorities()`

Also, it seems that we don't need a compound function here, tests could call `spawn_test_authorities()` followed by a new function, `spawn_fullnodes()`